pkg/services/internal/whatsonchain/dto/block_headers.go

```diff
@@ -0,0 +1,121 @@
+//
+// It includes core header fields, metadata about the block’s position and content,
+// and references to adjacent blocks.
+type BlockHeader struct {
+	// Hash is the block's unique identifier, computed as the double SHA-256 hash of the serialized block header.
+	Hash string `json:"hash"`
+
+	// Confirmations is the number of blocks added on top of this block.
+	// A higher number indicates deeper confirmation in the blockchain.
+	Confirmations uint `json:"confirmations"`
+
+	// Size is the total size of the block in bytes, including header and transactions.
+	Size uint `json:"size"`
+
+	// Height is the block’s position in the chain, starting from 0 (the genesis block).
+	Height uint `json:"height"`
+
+	// Version is the integer version number of the block.
+	// It indicates which consensus rules or features are in effect.
+	Version uint64 `json:"version"`
+
+	// VersionHex is the hexadecimal-encoded version field, matching the raw serialized representation.
+	VersionHex string `json:"versionHex"`
+
+	// MerkleRoot is the root hash of the Merkle tree formed by all transactions in the block.
+	// This value appears in the block header and proves transaction inclusion.
+	MerkleRoot string `json:"merkleroot"`
+
+	// Time is the block’s creation timestamp in Unix epoch seconds, set by the miner.
+	Time uint64 `json:"time"`
+
+	// Mediantime is the median timestamp of the previous 11 blocks.
+	// It is used for enforcing time-based rules like locktime.
+	Mediantime uint64 `json:"mediantime"`
+
+	// Nonce is a 32-bit number that miners iterate to find a hash meeting the difficulty target.
+	Nonce uint64 `json:"nonce"`
+
+	// Bits is the compact, encoded representation of the difficulty target in hexadecimal.
+	Bits string `json:"bits"`
+
+	// Difficulty is the actual mining difficulty as a floating-point number.
+	// It represents how hard it was to mine the block relative to the easiest possible difficulty.
+	Difficulty float64 `json:"difficulty"`
+
+	// Chainwork is a hex string representing the total cumulative proof-of-work in the chain up to and including this block.
+	Chainwork string `json:"chainwork"`
+
+	// PreviousBlockHash is the hash of the preceding block in the chain.
+	PreviousBlockHash string `json:"previousblockhash"`
+
+	// NextBlockHash is the hash of the succeeding block, if it exists.
+	// It may be empty if this block is the current chain tip.
+	NextBlockHash string `json:"nextblockhash"`
+
+	// NTx is the reported number of transactions in the block (may be a legacy or alias field).
+	NTx int `json:"nTx"`
+
+	// NumTx is the number of transactions in the block.
+	// It may be equal to or override NTx depending on the API source.
+	NumTx int `json:"num_tx"`
+}
+
+// ConvertToChainBlockHeader converts a BlockHeader into a *wdk.ChainBlockHeader used in chain processing.
+//
+// It parses the `Bits` field from a hexadecimal string to a uint64 integer.
+// Returns an error if the conversion fails.
```

Comment:
this is not needed - and could become obsolete if something change in the future (for example ChainBlockHeader would also use string)